core/controllers/story_editor.py

```diff
@@ -149,15 +145,6 @@ def delete(self, topic_id, story_id):
         if story is None:
             raise self.PageNotFoundException
 
-        topic = topic_services.get_topic_by_id(topic_id, strict=False)
```

Comment:
If you are removing this, make sure to add the corresponding check to acl_decorator.

Reply:
There is already a check in acl_decorators [here](https://github.com/oppia/oppia/blob/develop/core/controllers/acl_decorators.py#L2022) which checks if the topic exists or not. The condition is checked [here](https://github.com/oppia/oppia/blob/develop/core/controllers/acl_decorators.py#L2043)

Reply:
Yeah, but that's for topic rights, right? Although it is supposed to be linked to the existence of a topic, it would be better to have both checks there.

Reply:
Done

Reply:
Ok so on second thoughts if I add the second check to test the existence of a topic, the line with `get_topic_rights` will get untested as it can't be reached

Reply:
Reverted the changes!

Reply:
Reverted which changes? Shouldn't this deleted block be gone then?

Reply:
No I reverted the changes in acl_decorators. I deleted the above block as an exception will be rendered from the acl_decorators if an invalid topic_id is passed. See [here](https://github.com/oppia/oppia/blob/develop/core/controllers/acl_decorators.py#L2043)

Reply:
Then merge both checks into one in the acl_decorator, maybe using an OR.

Reply:
Similar explanatory comment here, please!

Reply:
Done and @brianrodri I shifted this check to the acl_decorators, so is there any need for a comment here? I did not remove any check